crates/iceberg/src/arrow/reader.rs

```diff
@@ -91,12 +98,15 @@ impl ArrowReader {
 
         Ok(try_stream! {
             while let Some(Ok(task)) = tasks.next().await {
-                let parquet_reader = file_io
-                    .new_input(task.data().data_file().file_path())?
+                let parquet_file = file_io
+                    .new_input(task.data().data_file().file_path())?;
+                let parquet_metadata = parquet_file.metadata().await?;
+                let parquet_reader =parquet_file
```

Comment:
Can we use `futures::try_join` here to run both of these futures simultaneously rather than sequentially?

```rust
let (parquet_metadata, parquet_reader) = try_join!(parquet_file.metadata(), parquet_file.reader())?;
```

Reply:
Fixed. Thanks for the suggestion.